07_boolean_2/main.go

```diff
@@ -15,56 +15,136 @@ E.g.:
+// the following checks:
+// - If the string is empty, return "empty"
+// - If the string is longer than the integer
+//   - If the integer is even, return "longer and even"
+//   - If the integer is odd, return "longer and odd"
+//
+// - If the string is shorter than the integer, return "shorter"
+func strIntCheck(str string, num int) {
+	// var stringLength = len(str)
+	// var integerLength = len(num)
+	if str == "" {
+		fmt.Println("empty")
+		// TODO find way to get int length
+		// } else if len(str) > len(num) {
+		if len(str)%2 == 0 {
+			fmt.Println("longer and even")
+		} else {
+			fmt.Println("longer and odd")
+		}
+	} else {
+		fmt.Println("shorter")
+	}
+}
 
-	// TODO: Write a function that takes 5 integers (grades) and returns the
-	// following:
-	// - If any of the grades is less than 0 or greater than 100, return
-	// "invalid grade".
-	// - If the average of the grades is greater than or equal to 90
-	//   - If no grade is below 80, return "A+"
-	//   - If no grade is below 70, return "A"
-	//   - If any grade is below 70, return "A-"
-	// - If the average of the grades is greater than or equal to 80, return
-	// "B"
-	//   - If no grade is below 80, return "B+"
-	//   - If no grade is below 70, return "B"
-	//   - If any grade is below 60, return "B-"
-	// - If the average of the grades is greater than or equal to 70, return
-	// "C"
-	//   - If no grade is below 70, return "C+"
-	//   - If no grade is below 60, return "C"
-	//   - If any grade is below 60, return "C-"
+// TODO: Write a function that takes 5 integers (grades) and returns the
+// following:
+// - If any of the grades is less than 0 or greater than 100, return
+// "invalid grade".
+// - If the average of the grades is greater than or equal to 90
+//   - If no grade is below 80, return "A+"
+//   - If no grade is below 70, return "A"
+//   - If any grade is below 70, return "A-"
+//
+// - If the average of the grades is greater than or equal to 80, return
+// "B"
+//   - If no grade is below 80, return "B+"
+//   - If no grade is below 70, return "B"
+//   - If any grade is below 60, return "B-"
+//
+// - If the average of the grades is greater than or equal to 70, return
+// "C"
+//   - If no grade is below 70, return "C+"
+//   - If no grade is below 60, return "C"
+//   - If any grade is below 60, return "C-"
+func calcGrade(grade1 int, grade2 int, grade3 int, grade4 int, grade5 int) {
+	var avg = (grade1 + grade2 + grade3 + grade4 + grade5) / 5
+	// var allGradesAnd = grade1 && grade2 && grade3 && grade4 && grade5
+	// var allGradesOr = grade1 || grade2 || grade3 || grade4 || grade5
+	// if grade.. > 100 || grade.. < 0 print invalid
+	if (grade1 < 0 || grade1 > 100) || (grade2 < 0 || grade2 > 100) || (grade3 < 0 || grade3 > 100) || (grade4 < 0 || grade4 > 100) || (grade5 < 0 || grade5 > 100) {
+		fmt.Println(("invalid grade"))
+		// If average grade  >= 90
+	} else if avg >= 90 {
+		if grade1 < 80 && grade2 < 80 && grade3 < 80 && grade4 < 80 && grade5 < 80 {
```

Comment:
how about you create a function called `anyValueUnder(...)` that returns true/false?

Reply:
maybe also `allValuesOver(...)`?

Reply:
Done